src/Microsoft.AspNet.OData.Shared/Query/AutoSelectExpandHelper.cs

```diff
@@ -175,6 +164,11 @@ private static bool HasAutoExpandProperty(this IEdmModel edmModel, IEdmStructure
                 {
                     if (IsAutoSelect(property, pathProperty, edmStructuredType, edmModel, querySettings))
                     {
+                        if (autoSelectProperties == null)
+                        {
+                            autoSelectProperties = new List<SelectModelPath>(1);
```

Comment:
I don't see the value in having to perform this `null` check every iteration. It probably makes more sense to just put line 160 back